python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -888,7 +889,10 @@ def test_sample(self):
 
         self.assertRaises(TypeError, lambda: self.spark.range(1).sample(seed="abc"))
 
-        self.assertRaises(IllegalArgumentException, lambda: self.spark.range(1).sample(-1.0))
+        self.assertRaises(
+            (IllegalArgumentException, SparkConnectException),
```

Comment:
No a big deal, I think we can catch `SparkConnectGrpcException` for more precious exception testing.

Reply:
Can we refine Spark Connect error to throw `IllegalArgumentException` similarly?

Reply:
> Can we refine Spark Connect error to throw `IllegalArgumentException` similarly?

Ah, this way sounds better. Let me address it soon!

Reply:
Just submitted https://github.com/apache/spark/pull/39783.

This should be simplified back to `self.assertRaises(IllegalArgumentException, lambda: self.spark.range(1).sample(-1.0))` after that change.